### ISSUE TITLE
App Crashes - Runtime Exception

### ISSUE THREAD

**Issue by wahibhaq** [nabinbhandari/Android-Permissions#2]
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.blah.blah.debug/com.nabinbhandari.android.permissions.PermissionsActivity}: java.lang.IllegalArgumentException: permission cannot be null or empty
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.IllegalArgumentException: permission cannot be null or empty
        at android.content.pm.PackageManager.buildRequestPermissionsIntent(PackageManager.java:3544)
        at android.app.Activity.requestPermissions(Activity.java:4381)
        at com.nabinbhandari.android.permissions.PermissionsActivity.onCreate(PermissionsActivity.java:72)
        at android.app.Activity.performCreate(Activity.java:7009)
        at android.app.Activity.performCreate(Activity.java:7000)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
```

I am observing on Permission Checks in my background service. While service is running and app is removed from recent apps, If I try to revoke permission it shows the dialog which makes total sense. But at the same time it also make the App alive in background and irrespective of pressing "Cancel" or "Ok", if I bring the app to foreground it crashes. In that state, I don't know why lib is getting null or empty permission. 

I am using app in production environment so it's kind of critical for me. if you have more questions to understand the problem I would be more than happy to help you out.

**Comment by nabinbhandari** [comment]
Hello, thank you for your feedback. I have also found some of some crashes from Google Play console. However, I am unable to reproduce the issue. Can you provide me with a minimal code and/or scenario that leads to the crash?

**Comment by wahibhaq** [comment]
Basically, it goes like this:

1) you are doing the permission check in a service (foreground) and it is running with a foreground notification.
2) you go to app's permissions in Settings and disable "Location" as in my case
3) System will kill the application but I have set Service to be `START_STICKY` in `onStartCommand()` so it will recreate it. 
4) I will see the rational dialog saying that this app needs location permission. Usually its quite fast and you see the dialog while you are on the app permission in Settings.

Now at this point, the app itself also appears in background and I think because of `PermissionActivity`. Problem occurs when you try to bring the app to foreground because it will crash and you will see the error in the logs which I mentioned in my first comment. 

I know it's quite an edge case but still a relevant one in my case where user can revoke the permission and I would like to show a dialog to make him aware of the consequences.

**Comment by wahibhaq** [comment]
I might have tried to fix it myself but lack extra time to invest so maybe there is something you can do about it otherwise I will find a workaround.

**Comment by nabinbhandari** [comment]
Hello, I have made an attempt to fix the issue. Can you please checkout the branch "temp" and verify if it has been solved?

To test:
- Checkout branch 'temp'.
- Run the task 'assembleRelease' of module 'permissions' in Gradle.
- Locate the AAR file in the build/outputs/aar
- Use the AAR file as dependency in your project instead of gradle dependency.

**Comment by wahibhaq** [comment]
THanks @nabinbhandari for the effort. I will try it out and will share my feedback.

**Comment by wahibhaq** [comment]
No man, unfortunately it is still getting the same error. I definitely used the `permissions-release.aar` file and used the module instead of the global dependency. 

```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.blah.blah.debug/com.nabinbhandari.android.permissions.PermissionsActivity}: java.lang.IllegalArgumentException: permission cannot be null or empty
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2927)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2988)
        at android.app.ActivityThread.-wrap14(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1631)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
     Caused by: java.lang.IllegalArgumentException: permission cannot be null or empty
        at android.content.pm.PackageManager.buildRequestPermissionsIntent(PackageManager.java:4054)
        at android.app.Activity.requestPermissions(Activity.java:4294)
        at com.nabinbhandari.android.permissions.PermissionsActivity.onCreate(PermissionsActivity.java:72)
        at android.app.Activity.performCreate(Activity.java:6942)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2880)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2988) 
        at android.app.ActivityThread.-wrap14(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1631) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6682) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410) 
```

Probably, I will let the service not start again once permissions are revoked but show a notification as a fallback plan.

**Comment by nabinbhandari** [comment]
at android.app.Activity.requestPermissions(Activity.java:4294)
    at com.nabinbhandari.android.permissions.PermissionsActivity.onCreate(PermissionsActivity.java:72)

I am sure you are still using the old code because the source code has changed and line of crash is still the same.

Perhaps you checked out the master branch instead of temp.

**Comment by wahibhaq** [comment]
Ahan, maybe I made a mistake. Let me try again.

**Comment by nabinbhandari** [comment]
The crash has been fixed. Thank you.
